int32 index
float64 force
float64 duration
---
bool ok
